publication-doi-commons/src/main/java/no/unit/nva/publication/doi/PublicationMapper.java

```diff
@@ -63,20 +69,36 @@ public PublicationMapping fromDynamodbStreamRecord(DynamodbStreamRecord streamRe
 
             if (acceptStreamViewTypes(streamViewType,
                 StreamViewType.NEW_AND_OLD_IMAGES, StreamViewType.OLD_IMAGE)) {
-                Publication oldPublication = fromDynamodbStreamRecordImage(dynamodb.getOldImage());
-                publicationMappingBuilder.withOldPublication(oldPublication);
+                maybeSetOldPublication(publicationMappingBuilder, dynamodb);
```

Comment:
`optionalSetOldPublication`  . `Maybe` is kotlins naming/version of `Optional`. The `if` check is needed because the mapper will throw unrecognized data illegal state exception if not. 

Unsure if we want data classes to have optional getters?

Reply:
After looking around I would name it just `setOldPublication`, you can keep maybe, but I find it confusing. Also if Maybe refers to Optionals, you do not use Optionals.

Reply:
Fixed by refactoring.